pandas/tests/frame/indexing/test_indexing.py

```diff
@@ -1332,12 +1338,22 @@ def test_loc_expand_empty_frame_keep_midx_names(self):
         )
         tm.assert_frame_equal(df, expected)
 
-    @pytest.mark.parametrize("val", ["x", 1])
-    @pytest.mark.parametrize("idxr", ["a", ["a"]])
-    def test_loc_setitem_rhs_frame(self, idxr, val):
+    @pytest.mark.parametrize(
+        "val, idxr, warn",
+        [
+            ("x", "a", None),
+            ("x", ["a"], None),
+            (1, "a", None),
+            (1, ["a"], FutureWarning),
```

Comment:
this seems weird to me.  the others also cast, right?

Reply:
agree that it's weird - they don't because the RHS of the assignment is a `DataFrame`, rather than a single value or a Series, so it goes down a different path. I've added comments to indicate this is todo